# HG changeset patch
# User Jan Beulich <jbeulich@suse.com>
# Date 1377594915 -7200
# Node ID c1b9dd3386633c9b1be16976dc608df76a2ea8d9
# Parent  6280dbacf66e26e17f1dae72b655a48ea0aac117
x86/Intel: add support for Haswell CPU models

... according to their most recent public documentation.

Signed-off-by: Jan Beulich <jbeulich@suse.com>
Acked-by: Keir Fraser <keir@xen.org>

diff -r 6280dbacf66e -r c1b9dd338663 xen/arch/x86/acpi/cpu_idle.c
--- a/xen/arch/x86/acpi/cpu_idle.c
+++ b/xen/arch/x86/acpi/cpu_idle.c
@@ -135,8 +135,10 @@ static void do_get_hw_residencies(void *
     case 0x3A:
     case 0x3E:
     /* Haswell */
-    case 0x3c:
+    case 0x3C:
+    case 0x3F:
     case 0x45:
+    case 0x46:
         GET_PC2_RES(hw_res->pc2);
         GET_CC7_RES(hw_res->cc7);
         /* fall through */
diff -r 6280dbacf66e -r c1b9dd338663 xen/arch/x86/hvm/vmx/vmx.c
--- a/xen/arch/x86/hvm/vmx/vmx.c
+++ b/xen/arch/x86/hvm/vmx/vmx.c
@@ -1815,7 +1815,7 @@ static const struct lbr_info *last_branc
         /* Ivy Bridge */
         case 58: case 62:
         /* Haswell */
-        case 60: case 69:
+        case 60: case 63: case 69: case 70:
             return nh_lbr;
             break;
         /* Atom */
diff -r 6280dbacf66e -r c1b9dd338663 xen/arch/x86/hvm/vmx/vpmu_core2.c
--- a/xen/arch/x86/hvm/vmx/vpmu_core2.c
+++ b/xen/arch/x86/hvm/vmx/vpmu_core2.c
@@ -909,7 +909,12 @@ int vmx_vpmu_initialise(struct vcpu *v, 
 
         case 0x3a: /* IvyBridge */
         case 0x3e: /* IvyBridge EP */
-        case 0x3c: /* Haswell */
+
+        /* Haswell: */
+        case 0x3c:
+        case 0x3f:
+        case 0x45:
+        case 0x46:
             ret = core2_vpmu_initialise(v, vpmu_flags);
             if ( !ret )
                 vpmu->arch_vpmu_ops = &core2_vpmu_ops;
